pkgs/development/libraries/onnxruntime/default.nix

```diff
@@ -150,8 +193,14 @@ stdenv.mkDerivation rec {
     "-Donnxruntime_BUILD_UNIT_TESTS=ON"
     "-Donnxruntime_ENABLE_LTO=ON"
     "-Donnxruntime_USE_FULL_PROTOBUF=OFF"
+    (lib.cmakeBool "onnxruntime_USE_CUDA" cudaSupport)
+    (lib.cmakeBool "onnxruntime_USE_NCCL" cudaSupport)
   ] ++ lib.optionals pythonSupport [
     "-Donnxruntime_ENABLE_PYTHON=ON"
+  ] ++ lib.optionals cudaSupport [
+    "-DFETCHCONTENT_SOURCE_DIR_CUTLASS=${cutlass}"
+    "-Donnxruntime_CUDNN_HOME=${cudaPackages.cudnn}"
+    "-DCMAKE_CUDA_ARCHITECTURES=${cudaArchitecturesString}"
   ];
 
   env = lib.optionalAttrs stdenv.cc.isClang {
```

Comment:
This may now be referring to a different stdenv than the one used for the build. I don't know of a good solution for this, but here's what I tried before: https://github.com/NixOS/nixpkgs/blob/5d4dd2a3645c048ead9b28a761e7e9d7644c078c/pkgs/development/libraries/opencv/4.x.nix#L90-L93

Reply:
I see. Replaced all the uses of stdenv in this way.